ec/ecdatasource/deploymenttemplates/schema.go

```diff
@@ -0,0 +1,173 @@
+		},
+	}
+}
+
+func deploymentTemplatesListSchema() schema.ListNestedAttribute {
+	return schema.ListNestedAttribute{
+		Description: "List of available deployment templates.",
+		Computed:    true,
+		NestedObject: schema.NestedAttributeObject{
+			Attributes: map[string]schema.Attribute{
+				"id": schema.StringAttribute{
+					Description: "The id of the deployment template.",
+					Computed:    true,
+				},
+				"name": schema.StringAttribute{
+					Description: "The name of the deployment template.",
+					Computed:    true,
+				},
+				"description": schema.StringAttribute{
+					Description: "The description of the deployment template.",
+					Computed:    true,
+				},
+				"min_stack_version": schema.StringAttribute{
+					Description: "The minimum stack version that can used with this deployment template.",
+					Computed:    true,
+				},
+				"deprecated": schema.BoolAttribute{
+					Description: "Outdated templates are marked as deprecated, but can still be used.",
+					Computed:    true,
+				},
+				"elasticsearch":       elasticsearchSchema(),
+				"kibana":              statelessSchema(),
+				"enterprise_search":   statelessSchema(),
+				"apm":                 statelessSchema(),
+				"integrations_server": statelessSchema(),
+			},
+		},
+	}
+}
+
+func elasticsearchSchema() schema.SingleNestedAttribute {
+	return schema.SingleNestedAttribute{
+		Description: "Defines the default configuration for Elasticsearch.",
+		Computed:    true,
+		Attributes: map[string]schema.Attribute{
+			"hot":          topologySchema(),
+			"coordinating": topologySchema(),
+			"master":       topologySchema(),
+			"warm":         topologySchema(),
+			"cold":         topologySchema(),
+			"frozen":       topologySchema(),
+			"ml":           topologySchema(),
+		},
+	}
+}
+
+func topologySchema() schema.SingleNestedAttribute {
+	return schema.SingleNestedAttribute{
+		Description: "Defines the default configuration for each topology.",
+		Computed:    true,
+		Attributes: map[string]schema.Attribute{
+			"instance_configuration_id": schema.StringAttribute{
+				Computed: true,
+			},
+			"instance_configuration_version": schema.NumberAttribute{
+				Computed: true,
+			},
+			"default_size": schema.StringAttribute{
+				Computed: true,
+			},
+			"available_sizes": schema.ListAttribute{
+				ElementType: types.StringType,
+				Computed:    true,
+			},
+			"size_resource": schema.StringAttribute{
+				Computed: true,
+			},
+			"autoscaling": autoscalingSchema(),
+		},
+	}
+}
+
+func autoscalingSchema() schema.SingleNestedAttribute {
+	return schema.SingleNestedAttribute{
+		Computed: true,
+		Attributes: map[string]schema.Attribute{
+			"max_size_resource": schema.StringAttribute{
+				Computed: true,
+			},
+			"max_size": schema.StringAttribute{
+				Computed: true,
+			},
+			"min_size_resource": schema.StringAttribute{
+				Computed: true,
+			},
+			"min_size": schema.StringAttribute{
+				Computed: true,
+			},
+		},
```

Comment:
Does it make sense to include tier autoscaling override too?

Reply:
Which one do you mean? This `autoscalingSchema` is currently set for every tier/topology-element

Reply:
https://github.com/elastic/cloud-sdk-go/blob/master/pkg/models/elasticsearch_cluster_topology_element.go#L51

Reply:
I added them now.